### PR TITLE
pm: device: move device busy APIs to pm subsystem

### PR DESCRIPTION
**Pull request by gmarull**
Summary:

```
pm: device: move device busy APIs to pm subsystem

The following device busy APIs:

- device_busy_set()
- device_busy_clear()
- device_busy_check()
- device_any_busy_check()

were used for device PM, so they have been moved to the pm subsystem.
This means they are now prefixed with `pm_` and are defined in
`pm/device.h`.

If device PM is not enabled dummy functions are now provided that do
nothing or return `-ENOSYS`, meaning that the functionality is not
available.
```
```
tests: kernel: device: remove redundant busy set and clear

The test_dummy_device test has nothing to do with device busy testing,
so remove the calls to pm_device_busy_set/pm_device_busy_clear.
```
```
pm: device: improve usage of the atomic flags field

- Use an enum for the available flags
- Use ATOMIC_DEFINE macro to declare the flag bit field
- Improve naming
```

### CODE REVIEW

**Review by dcpleung — APPROVED** [review]

**Review by ceolin — DISMISSED** [review]

**Review by nashif — CHANGES_REQUESTED** [review]
Need to consider/agree if deprecation is needed here before we move forward with this.

**Review by tbursztyka — APPROVED** [review]
Aside from @nashif's comment, lgtm.

**Comment by gmarull** [comment]
> Need to consider/agree if deprecation is needed here before we move forward with this.

@nashif After a quick discussion yesterday, we think these calls should be considered part of PM (proof is that they use the PM atomic bit field to operate). Since PM API is not stable, deprecation should not be considered for this case.

File: subsys/pm/device.c [hunk 1]
```diff
@@ -149,3 +149,39 @@ int pm_device_state_get(const struct device *dev,
...
+				    PM_DEVICE_ATOMIC_FLAGS_BUSY_BIT)) {
+			return -EBUSY;
+		}
+		++dev;
+	}
+
+	return 0;
+}
+
+int pm_device_busy_check(const struct device *dev)
```
Comment (ceolin) [inline subsys/pm/device.c:167]: would be better and more aligned with the rest of the API if it was `bool pm_device_is_busy()`

**Review by ceolin — APPROVED** [review]

File: subsys/pm/device.c (hunk 1 above)
Comment (gmarull) [reply subsys/pm/device.c]: agreed, adjusted it (also the `any` variant). I have also updated release notes with the changes.

Comment (gmarull) [reply subsys/pm/device.c]: well, I've checked it again: the functions return `int`, not `bool`, so I'm not sure about the change in that case. I'd only use `_is_` pattern for boolean checks. Adding DNM until this point is clarified.

**Review by erwango — APPROVED** [review]

File: subsys/pm/device.c (hunk 1 above)
Comment (gmarull) [reply subsys/pm/device.c]: Missed that suggestion included move to `bool`, so it should be ready now.

**Comment by gmarull** [comment]
@nashif should be ready now

**Review by nashif — APPROVED** [review]

**Pull request merged**